File: go/rotational-cipher/rotational_cipher.go
package rotationalcipher

// RotationalCipher encodes a string with the given rotation
func RotationalCipher(s string, rot int) (res string) {
	for _, r := range s {
		if r >= 'a' && r <= 'z' {
			res += string(((r + rune(rot) - 'a') % 26) + 'a')
		} else if r >= 'A' && r <= 'Z' {
			res += string(((r + rune(rot) - 'A') % 26) + 'A')
		} else {
			res += string(r)
		}
	}
	return
}

File: typescript/rotational-cipher/rotational-cipher.ts
export default class RotationalCipher {
  static rotate(s: string, rot: number) {
    let res: string = '';
    const A: number = 'A'.charCodeAt(0);
    const Z: number = 'Z'.charCodeAt(0);
    for (const c of s) {
      const upperCharCode: number = c.toUpperCase().charCodeAt(0);
      const letterOffset: number = upperCharCode - A;
      if (upperCharCode >= A && upperCharCode <= Z) {
        const shift: number = (letterOffset + rot) % 26 - letterOffset;
        res += String.fromCharCode(c.charCodeAt(0) + shift);
      } else {
        res += c;
      }
    }
    return res;
  }
}
